packages/perseus/src/widgets/image/image.tsx

```diff
@@ -29,6 +30,21 @@ export const ImageComponent = (props: ImageWidgetProps) => {
     const context = React.useContext(PerseusI18nContext);
     const imageUpgradeFF = isFeatureOn({apiOptions}, "image-widget-upgrade");
 
+    const [largerImageSize, setLargerImageSize] = React.useState<
+        [number, number]
+    >([backgroundImage.width || 0, backgroundImage.height || 0]);
+
+    React.useEffect(() => {
+        // Wait to figure out what the original size of the image is.
+        // Use whichever is larger between the original image size and the
+        // saved background image size for zooming.
+        Util.getImageSizeModern(backgroundImage.url!).then((size) => {
+            if (size[0] > (backgroundImage.width || 0)) {
+                setLargerImageSize(size);
+            }
+        });
+    }, [backgroundImage]);
```

Comment:
@jeremywiebe Do you know if this implementation has any issues?

I'm wondering if there's a reason we decide to save the image size in the widget data rather than use `Util.getImageSizeModern` on load. It helped me implement image resizing, but now I'd like to have both values (original size and preferred size). It would be nice to be able to use the util function to get the original size on load rather than have to save both sizes in the data.

Reply:
I recall Nicole working on that Util.getImageSizeModern function but I don’t know the history of why we captured the size.

My guess is that it allows us to present a placeholder of the right size while we load the image… and then when we render the image, there’s no page jank because the placeholder consumed the exact same dimensions as the image.

Reply:
Okay, cool. I think that means the implementation in this PR should be fine. If it causes any loading or timing issues in the future (which I don't anticipate), we can revert this.